console/lib/subscribe.dart

```diff
@@ -0,0 +1,151 @@
+    void fetchCommunities(BuildContext context) {
+        final authOptions = [authn.DeeppoolAuthzCache.bearer(context)];
+        
+        community_api.search(
+            options: [httpx.Accept.json, ...authOptions],
+            limit: 100
+        ).then((response) {
+            setState(() {
+                communities = response.items;
+                _cause = const SizedBox();
+            });
+        }).catchError((cause) {
+            setState(() {
+                _cause = ds.Error.offline(
+                    cause,
+                    onTap: _clearCause,
+                );
+            });
+        }, test: ds.ErrorTests.offline)
+        .catchError((cause) {
+            setState(() {
+                _cause = ds.Error.connectivity(
+                    cause,
+                    onTap: _clearCause,
+                );
+            });
+        }, test: ds.ErrorTests.connectivity)
+        .catchError((cause) {
+            setState(() {
+                _cause = ds.Error.unknown(
+                    cause,
+                    onTap: _clearCause,
+                );
+            });
+        })
+        .whenComplete(() {
+            setState(() {
+                loading = false;
+            });
+        });
+    }
+
+    @override
+    Widget build(BuildContext context) {
+        return ds.Loading(
+            key: ValueKey(communities.length), 
+            loading: loading, 
+            cause: _cause,
+            modals.Node(
+                ds.Full(
+                    communities.isEmpty && !loading && _cause is SizedBox
+                        ? Center(child: Text('No communities found'))
+                        : ListView.builder(
+                            itemCount: communities.length,
+                            itemBuilder: (context, index) {
+                                final community = communities[index];
+                                // Encode the full community information as JSON in the QR code
+                                final qrData = jsonEncode(community.toProto3Json());
+                                return Card(
+                                    child: Padding(
+                                        padding: const EdgeInsets.all(8.0),
+                                        child: Row(
+                                            children: [
+                                                Expanded(
+                                                    child: Column(
+                                                        crossAxisAlignment: CrossAxisAlignment.start,
+                                                        children: [
+                                                            Text(
+                                                                community.domain, 
+                                                                style: Theme.of(context).textTheme.titleMedium
```

Comment:
you probably want to pull the theme retrieval up as a final variable to limit the number of times you walk the tree.
```suggestion
                                                                style: theme.textTheme.titleMedium
```